### PR TITLE
List now works with one/no items

### PR DESCRIPTION
**Pull request by Matt-Butler**
### Summary
Multi select / Single select lists were unable to render with zero or one items. This pr fixes that issue.

https://github.com/cerner/terra-core/issues/425

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

### CODE REVIEW

File: packages/terra-list/src/MultiSelectList.jsx [hunk 1]
```diff
@@ -111,6 +111,18 @@ class MultiSelectList extends React.Component {
...
+
+    if (items.constructor !== Array) {
+      const wrappedOnClick = this.wrappedOnClickForItem(items, 0);
+      const wrappedOnKeyDown = this.wrappedOnKeyDownForItem(items, 0);
+      const newProps = this.newPropsForItem(items, 0, wrappedOnClick, wrappedOnKeyDown);
+
+      return React.cloneElement(items, newProps);
+    }
+
     return items.map((item, index) => {
```
Comment (emilyrohrbough) [inline packages/terra-list/src/MultiSelectList.jsx]: Would this do the same thing:
`return React.Children.map(items, (item, index) => {`

Comment (emilyrohrbough) [inline packages/terra-list/src/MultiSelectList.jsx]: That is the change we made in TableRows to handle the same issue.

Comment (Matt-Butler) [inline packages/terra-list/src/MultiSelectList.jsx]: Cool, I'll try that out. It is a lot more elegant than my approach

File: packages/terra-list/src/MultiSelectList.jsx [hunk 2]
```diff
@@ -111,15 +111,16 @@ class MultiSelectList extends React.Component {
   cloneChildItems(items) {
     const disableUnselectedItems = this.state.selectedIndexes.length >= this.validatedMaxCount();
 
-    return items.map((item, index) => {
-      const wrappedOnClick = this.wrappedOnClickForItem(item, index);
-      const wrappedOnKeyDown = this.wrappedOnKeyDownForItem(item, index);
-      const newProps = this.newPropsForItem(item, index, wrappedOnClick, wrappedOnKeyDown, disableUnselectedItems);
```
Comment (StephenEsser) [inline packages/terra-list/src/MultiSelectList.jsx]: Was `disableUnselectedItems` intentionally removed?

Comment (Matt-Butler) [inline packages/terra-list/src/MultiSelectList.jsx]: I have no idea what you are talking about. If I did remove disableUnselectedItems, I would have amended that commit and no one would be the wiser :)

**Comment by Matt-Butler** [comment]
@mmalaker @neilpfeiffer I accidently merged this without your +1s. 

I updated the doc site with the examples from this PR:
http://engineering.cerner.com/terra-core/#/tests/list-tests
http://engineering.cerner.com/terra-core/#/tests/single-select-list-tests
http://engineering.cerner.com/terra-core/#/tests/multi-select-list-tests

The bottom two examples on each page. The one item or no items links.

**Comment by mmalaker** [comment]
Functional verification completed. All examples and test pages display and function correctly.

**Pull request merged**